### PR TITLE
Make ZHA config panel device oriented

### PR DESCRIPTION
**Pull request by dmulcahey**
This cleans up the ZHA config panel a bit and makes it device oriented. It removes entities from all ZHA operations and does everything on a device level. 

**DO NOT MERGE WITHOUT:** https://github.com/home-assistant/home-assistant/pull/20990

### CODE REVIEW

File: src/panels/config/zha/zha-node.ts [hunk 1]
```diff
@@ -62,13 +54,20 @@ export class ZHANode extends LitElement {
...
       _selectedNodeIndex: {},
       _selectedNode: {},
+      _entities: {},
       _serviceData: {},
-      _selectedEntity: {},
+      _nodes: {},
     };
   }
 
+  public connectedCallback(): void {
```
Comment (balloob) [inline src/panels/config/zha/zha-node.ts]: You want to do initial fetching in `firstUpdated()`

File: src/panels/config/zha/zha-device-card.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            <dd class="info">${this.device!.ieee}</dd>
+            <dt class="label">Quirk applied:</dt>
+            <dd class="info">${this.device!.quirk_applied}</dd>
+            <dt class="label">Quirk:</dt>
+            <dd class="info">${this.device!.quirk_class}</dd>
+          </dl>
+        </div>
+
+        <div class="device-entities">
+          ${this.device!.entities.map(
```
Comment (balloob) [inline src/panels/config/zha/zha-device-card.ts]: Can you add a message if there are 0 entities? Or is that impossible?

File: src/panels/config/zha/zha-device-card.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            (entity) => html`
+              <paper-icon-item @click="${
```
Comment (balloob) [inline src/panels/config/zha/zha-device-card.ts]: To make this more readable, you can put the `@click` on it's own line.

File: src/panels/config/zha/zha-device-card.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+import "../../../components/entity/state-badge";
+import { ZHADevice } from "../../../data/zha";
+
+class ZHADeviceCard extends LitElement {
+  public hass?: HomeAssistant;
+  public narrow?: boolean;
+  private device?: ZHADevice;
+
+  constructor() {
```
Comment (balloob) [inline src/panels/config/zha/zha-device-card.ts]: Not needed

File: src/panels/config/zha/zha-device-card.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+import "@polymer/paper-listbox/paper-listbox";
+import { fireEvent } from "../../../common/dom/fire_event";
+import { haStyle } from "../../../resources/ha-style";
+import { HomeAssistant } from "../../../types";
+
+import "../../../components/entity/state-badge";
+import { ZHADevice } from "../../../data/zha";
+
+class ZHADeviceCard extends LitElement {
+  public hass?: HomeAssistant;
```
Comment (balloob) [inline src/panels/config/zha/zha-device-card.ts]: We are super cool now, you can do this:

```suggestion
  @property() public hass?: HomeAssistant;
```

And then drop `static get properties()`

See #2711

File: src/panels/config/zha/zha-node.ts [hunk 6]
```diff
@@ -186,50 +178,27 @@ export class ZHANode extends LitElement {
...
-      return [];
-    }
-  }
-
-  private _onEntitySelected(
-    entitySelectedEvent: HASSDomEvent<ZHAEntitySelectedParams>
-  ): void {
-    this._selectedEntity = entitySelectedEvent.detail.entity;
+  private async _fetchDevices() {
+    this._nodes = await fetchDevices(this.hass!);
```
Comment (balloob) [inline src/panels/config/zha/zha-node.ts]: Assigning twice _feels_ a bit weird, I would prefer:

```ts
    this._nodes = (await fetchDevices(this.hass!)).sort((a, b) => { … });
```

However, it's more like a nitpick, ignore if you want.

**Review by balloob — APPROVED** [review]
Can be merged when comments addressed.

File: src/panels/config/zha/zha-device-card.ts (hunk 2 above)
Comment (dmulcahey) [reply src/panels/config/zha/zha-device-card.ts]: every device will have at least one

File: src/panels/config/zha/zha-node.ts [hunk 7]
```diff
@@ -62,13 +55,31 @@ export class ZHANode extends LitElement {
...
+      _nodes: {},
     };
   }
 
+  public firstUpdated(changedProperties: PropertyValues): void {
+    super.firstUpdated(changedProperties);
+    if (this._nodes.length === 0) {
+      this._fetchDevices();
+    }
+    this.addEventListener("hass-service-called", (ev) =>
```
Comment (dmulcahey) [inline src/panels/config/zha/zha-node.ts]: @balloob I had to make a change here to clear selection and fetch fresh device list after remove is called. Just want to make sure the change is good before I merge this and the backend PR.

**Comment by dmulcahey** [comment]
All comments addressed and a new question added.

File: src/panels/config/zha/zha-node.ts (hunk 7 above)
Comment (balloob) [reply src/panels/config/zha/zha-node.ts]: If a remove is successful, just remove the device on the client side instead of refreshing devices?

Comment (dmulcahey) [reply src/panels/config/zha/zha-node.ts]: I can do that. Thx

Comment (balloob) [reply src/panels/config/zha/zha-node.ts]: Going to merge this now so I can do a build, you can do it in another PR.

**Review by balloob — APPROVED** [review]

**Pull request merged**